### PR TITLE
Add ability to create a new collection from the users page. Change ac…

### PR DESCRIPTION
**Pull request by nimkamp**
…tion route in new collection component to specify where we want to post to instead of default. Create modal for new collection form

### CODE REVIEW

**Comment by scottPlusPlus** [comment]
In general great:

1. Please add a screen recording demonstrating that the feature works.
2. I noticed that if you try to give the new collection an id that is already taken, the resulting experience is confusing.   Let's check for an error coming from the request and if so display it on the page

File: app/routes/me.tsx [hunk 1]
```diff
@@ -27,13 +30,38 @@ export default function UserPage() {
...
+  }
+
   return (
     <div className="flex h-full min-h-screen flex-col">
       <header className="flex items-center justify-between bg-slate-800 p-4 text-white">
         <h1 className="text-3xl font-bold">
           Hello
         </h1>
         <p>{user.email}</p>
+        <div>
```
Comment (scottPlusPlus) [inline app/routes/me.tsx]: can we break the modal out into its own component?   (does NOT need to be in its own file, just stick it at the bottom of the file for now)

Comment (nimkamp) [reply app/routes/me.tsx]: Just completed this.

**Comment by nimkamp** [comment]
Screenshot 2023-07-30 at 3 57 23 PM

**Comment by nimkamp** [comment]
https://github.com/scottPlusPlus/uscout/assets/5142680/d115f529-5c15-4a86-a681-e91406747bc2

**Pull request merged**